src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.java

```diff
@@ -83,18 +96,19 @@ public void visitToken(DetailAST ast) {
                 && exprStart.getType() == TokenTypes.LITERAL_NULL) {
                 log(ident, MSG_KEY, ident.getText(), "null");
             }
-
-            final int primitiveType = type.getFirstChild().getType();
-            if (primitiveType == TokenTypes.LITERAL_BOOLEAN
-                && exprStart.getType() == TokenTypes.LITERAL_FALSE) {
-                log(ident, MSG_KEY, ident.getText(), "false");
-            }
-            if (isNumericType(primitiveType) && isZero(exprStart)) {
-                log(ident, MSG_KEY, ident.getText(), "0");
-            }
-            if (primitiveType == TokenTypes.LITERAL_CHAR
-                && isZeroChar(exprStart)) {
-                log(ident, MSG_KEY, ident.getText(), "\\0");
+            if (!onlyObjectReferences) {
```

Comment:
Why not move everything in this block to a new method called `validateNonObjects`?